content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,563 @@
+`female`, we should obtain the vector corresponding to `queen`. That being said, those kind of representation usually are in dimension `300`, which
+makes it impossible for humans to reason about them. Still, the idea is there but in a larger dimension space.
+
+So our LSTM's input and hidden state dimensions will be of the same size as the vectors of embedded words. 
+For the present purpose, we will use the
+[French pre trained](https://fasttext.cc/docs/en/crawl-vectors.html) fastText embeddings of dimension `300`. 
+
+### The Pytorch Model
+
+Let us first import all the required packages.
+
+```python
+import os
+import pickle
+
+import fasttext
+import fasttext.util
+import requests
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from poutyne import set_seeds
+from poutyne.framework import Experiment
+from torch.nn.functional import cross_entropy
+from torch.nn.utils.rnn import pad_packed_sequence, pack_padded_sequence, pad_sequence
+from torch.utils.data import DataLoader
+```
+
+Now, let's create a single (i.e. one layer) unidirectional LSTM wit `input_size` and `hidden_size` of `300`. We 
+will explore later on the effect of stacking more layers and of using a bidirectional approach.
+
+> See [here](https://discuss.pytorch.org/t/could-someone-explain-batch-first-true-in-lstm/15402) the explanation why we use the `batch_first` argument.
+
+```python
+dimension = 300
+num_layer = 1
+bidirectional = False
+
+lstm_network = nn.LSTM(input_size=dimension,
+                       hidden_size=dimension,
+                       num_layers=num_layer,
+                       bidirectional=bidirectional,
+                       batch_first=True)
+```
+
+
+## Fully-connected Layer
+Since the output of the LSTM network is of dimension `300`, we will use a fully-connected layer to map this large dimension space into
+a smaller one of dimension equal to that of the tag space (i.e. number of tags to predict), that is 8. 
+
+Finally, since we want to predict the most probable tokens, we will apply the softmax function on this layer
+(see [here](https://en.wikipedia.org/wiki/Softmax_function) if softmax does not ring a bell).
+
+
+```python
+input_dim = dimension #the output of the LSTM
+tag_dimension = 8
+
+fully_connected_network = nn.Linear(input_dim, tag_dimension)
+```
+
+## Training Constants
+
+Now, let's set our training constants. We first have the CUDA device used for training (using a CPU is way too long, 
+if you don't have one, you can use a Google Colab notebook). 
+
+Secondly, we set the batch size (i.e. the number of elements to see before updating the model), the learning rate for the optimizer
+and the number of epochs.
+
+```python
+device = torch.device("cuda:0")
+
+batch_size = 128
+lr = 0.1
+
+epoch_number = 10
+```
+
+We also need to set Pythons's, NumPy's and PyTorch's seeds using Poutyne function so that our training is (almost) reproducible.
```

Comment:
```suggestion
We also need to set Pythons's, NumPy's and PyTorch's seeds using the Poutyne function so that our training is (almost) reproducible.
```